OSDOCS-14611 created z-stream release notes for 4.18.13 (#93209)
openshift/openshift-docs: release_notes/ocp-4-18-release-notes.adoc

For any {product-title} release, always review the instructions on xref:../updating/updating_a_cluster/updating-cluster-web-console.adoc#updating-cluster-web-console[updating your cluster] properly.
====

// 4.18.13
[id="ocp-4-18-13_{context}"]

> === RHSA-2025:4712 - {product-title} {product-version}.12 bug fix update and security update

Reviewer: ```suggestion
=== RHSA-2025:4712 - {product-title} {product-version}.13 bug fix update and security update
```

Issued: 08 May 2025

> {product-title} release {product-version}.12 is now available. The list of bug fixes that are included in the update is documented in the link:https://access.redhat.com/errata/RHSA-2025:4712[RHSA-2025:4712] advisory. The RPM packages that are included in the update are provided by the link:https://access.redhat.com/errata/RHBA-2025:4714[RHBA-2025:4714] advisory.

Reviewer: ```suggestion
{product-title} release {product-version}.13 is now available. The list of bug fixes that are included in the update is documented in the link:https://access.redhat.com/errata/RHSA-2025:4712[RHSA-2025:4712] advisory. The RPM packages that are included in the update are provided by the link:https://access.redhat.com/errata/RHBA-2025:4714[RHBA-2025:4714] advisory.
```

Author: Done. Also changed date from 08 May 2025 to 14 May 2025

Space precluded documenting all of the container images for this release in the advisory.

You can view the container images in this release by running the following command:

[source,terminal]
----
$ oc adm release info 4.18.13 --pullspecs
----

[id="ocp-4-18-13-known-issues_{context}"]
==== Known issues

> * When a pod uses the CNI plugin for DHCP address assignment, in conjunction with other CNI plugins, the pod's network interface might be unexpectedly deleted. As a result, when the pod's DHCP lease expires, the DHCP proxy enters a loop when trying to re-create a new lease, leading to the node becoming unresponsive. There is currently no known workaround. (link:https://issues.redhat.com/browse/OCPBUGS-55354[OCPBUGS-55354])

Author: Done

[id="ocp-4-18-13-bug-fixes_{context}"]
==== Bug fixes

> * Previously, unsorted image stream names in the `Progressing` condition were causing unnecessary updates. This led to excessive user updates and potential performance degradation. With this release, the failing image imports are sorted within the `activeImageStreams` function improving sample operator efficiency, reducing unnecessary updates, and enhancing overall performance. (link:https://issues.redhat.com/browse/OCPBUGS-55783[OCPBUGS-55783])

Reviewer: ```suggestion
* Previously, unsorted image stream names in the `Progressing` condition caused unnecessary updates. This led to excessive user updates and potential performance degradation. With this release, the failing image imports are sorted within the `activeImageStreams` function improving sample operator efficiency, reducing unnecessary updates, and enhancing overall performance. (link:https://issues.redhat.com/browse/OCPBUGS-55783[OCPBUGS-55783])
```

Author: Done

> * Previously, unsorted image stream names in the `Progressing` condition caused unnecessary updates. This led to excessive user updates and potential performance degradation. With this release, the failing image imports are sorted within the `activeImageStreams` function improving sample operator efficiency, reducing unnecessary updates, and enhancing overall performance. (link:https://issues.redhat.com/browse/OCPBUGS-55783[OCPBUGS-55783])

Reviewer 2: ```suggestion
* Previously, unsorted image stream names in the `Progressing` condition caused unnecessary updates. This led to excessive user updates and potential performance degradation. With this release, the failing image imports are sorted within the `activeImageStreams` function. This change improves Cluster Samples Operator efficiency, reduces unnecessary updates, and enhances overall performance. (link:https://issues.redhat.com/browse/OCPBUGS-55783[OCPBUGS-55783])
```

> * Previously, an operator updated the 'Progressing' condition's `lastTransitionTime` even when the condition's status did not actually change. This led to potential installation errors and perceived instability for end users. With this release, the operator is prevented from updating the `lastTransitionTime` unless there is a status change, enhancing operator stability, minimizing installer errors, and ensuring a smoother user experience. (link:https://issues.redhat.com/browse/OCPBUGS-55782[OCPBUGS-55782])

Reviewer 2: ```suggestion
* Previously, an Operator updated the `Progressing` condition's `lastTransitionTime` value even when the condition's status did not actually change. This led to potential installation errors and perceived instability for end users. With this release, the Operator is prevented from updating the `lastTransitionTime` value unless there is a status change. This update enhances Operator stability, minimizes installation errors, and ensures a smoother user experience. (link:https://issues.redhat.com/browse/OCPBUGS-55782[OCPBUGS-55782])
```

Author: @jeana-redhat Made the changes

> * Previously, an Operator updated the 'Progressing' condition's `lastTransitionTime` value even when the condition's status did not actually change. This led to potential installation errors and perceived instability for end users. With this release, the Operator is prevented from updating the `lastTransitionTime` value unless there is a status change. This enhances Operator stability, minimizes installer errors, and ensures a smoother user experience. (link:https://issues.redhat.com/browse/OCPBUGS-55782[OCPBUGS-55782])

Reviewer 2: ```suggestion
* Previously, an Operator updated the `Progressing` condition's `lastTransitionTime` value even when the condition's status did not actually change. This led to potential installation errors and perceived instability for end users. With this release, the Operator is prevented from updating the `lastTransitionTime` value unless there is a status change. This enhances Operator stability, minimizes installer errors, and ensures a smoother user experience. (link:https://issues.redhat.com/browse/OCPBUGS-55782[OCPBUGS-55782])
```

> * Previously, the sample operator watched all of the 'ClusterOperators' on any cluster operator where changes to the cluster triggered the syncloop to run. The resulted in excessive and unnecessary syncloop triggers negatively impacting overall performance. With this release, the 'ClusterOperator' watch was modified to filter for changes only to the 'openshift-samples' 'ClusterOperator', reducing unnecessary syncloops leading to improved performance. (link:https://issues.redhat.com/browse/OCPBUGS-55781[OCPBUGS-55781])

Author: Made all suggested changes

> * Previously, the sample Operator detected all of the 'ClusterOperators' on any cluster Operator where changes to the cluster triggered the sync loop to run. This change resulted in excessive and unnecessary sync loop triggers that negatively impacted the overall performance. With this release, the 'ClusterOperator' watch was modified to filter for changes only to the 'openshift-samples' 'cluster Operator', reducing unnecessary sync loops and improving performance. (link:https://issues.redhat.com/browse/OCPBUGS-55781[OCPBUGS-55781])

Reviewer 2: I don't know what 
```suggestion
* Previously, the Cluster Samples Operator watched all cluster Operators in the cluster, which triggered the Cluster Samples Operator sync loop to run unnecessarily. This behavior negatively impacted overall performance. With this release, the Cluster Samples Operator only watches specific cluster Operators. (link:https://issues.redhat.com/browse/OCPBUGS-55781[OCPBUGS-55781])
```

Author: @jeana-redhat Made the changes

> * Previously, when adding a node in a disconnected environment, private registry images were inaccessible for the `oc adm node-image` command. As a result, issues with pulling images prevented adding a node. This error only occurs if the cluster is initially installed with an installation program binary downloaded from mirror.openshift.com. The issue is resolved in this release. (link:https://issues.redhat.com/browse/OCPBUGS-55449[OCPBUGS-55449])

Reviewer 2: ```suggestion
* Previously, when adding a node in a disconnected environment, private registry images were inaccessible for the `oc adm node-image` command. As a result, issues with pulling images prevented adding a node. This error only occurs if the cluster is initially installed with an installation program binary downloaded from `\mirror.openshift.com`. The issue is resolved in this release. (link:https://issues.redhat.com/browse/OCPBUGS-55449[OCPBUGS-55449])
```

> * Previously, when adding a node in a disconnected environment, private registry images were inaccessible for the `oc adm node-image` command. As a result, issues with pulling images prevented adding a node. This error only occurs if the cluster is initially installed with an installation program binary downloaded from `\mirror.openshift.com`. The issue is resolved in this release. (link:https://issues.redhat.com/browse/OCPBUGS-55449[OCPBUGS-55449])

Reviewer 2: @wgabor0427 this my fault: the `\ at the front of the URL is to keep it from making that into a hyperlink. But it looks like it doesn't actually recognize this as a URL, so instead it's rendering the \ in text. It actually only needs this, for this particular URL:
```suggestion
* Previously, when adding a node in a disconnected environment, private registry images were inaccessible for the `oc adm node-image` command. As a result, issues with pulling images prevented adding a node. This error only occurs if the cluster is initially installed with an installation program binary downloaded from `mirror.openshift.com`. The issue is resolved in this release. (link:https://issues.redhat.com/browse/OCPBUGS-55449[OCPBUGS-55449])
```

Sorry for adding churn here!

Author: @jeana-redhat No worried. Removed \

Reviewer 2: thanks! LGTM

> * Currently, when adding a node in a disconnected environment, private registry images are inaccessible for the 'oc adm node-image' command, causing node addition failure due to image pull issues. This error only occurs if the cluster is initially installed with an installer binary downloaded from mirror.openshift.com. With this release, a fix has been implemented enabling successful image pull and node creation in disconnected environments. (link:https://issues.redhat.com/browse/OCPBUGS-55449[OCPBUGS-55449])

Author: done

> * Previously, when adding a node in a disconnected environment, private registry images are inaccessible for the `oc adm node-image` command, causing node addition failure due to image pull issues. This error only occurs if the cluster is initially installed with an installation program binary downloaded from mirror.openshift.com. With this release, a fix has been implemented resulting in a successful image pull and node creation in disconnected environments. (link:https://issues.redhat.com/browse/OCPBUGS-55449[OCPBUGS-55449])

Reviewer 2: ```suggestion
* Previously, when adding a node in a disconnected environment, private registry images were inaccessible for the `oc adm node-image` command. As a result, issues with pulling images prevented adding a node. This error only occurs if the cluster is initially installed with an installation program binary downloaded from `\mirror.openshift.com`. The issue is resolved in this release. (link:https://issues.redhat.com/browse/OCPBUGS-55449[OCPBUGS-55449])
```

> * Currently, there is an issue in the image reference digest calculation leading to a failed container creation based on the schemaVersion 1 image. This prevents new deployment creations. With this release, the image digest calculation has been fixed and new operators can be installed. (link:https://issues.redhat.com/browse/OCPBUGS-55435[OCPBUGS-55435])

Reviewer: ```suggestion
* Previously, there was an issue in the image reference digest calculation that led to a failed container creation based on the `schemaVersion 1` image. This prevented new deployment creations. With this release, the image digest calculation is fixed and new Operators can be installed. (link:https://issues.redhat.com/browse/OCPBUGS-55435[OCPBUGS-55435])
```

Author: Done

> * Previously, there was an issue in the image reference digest calculation that led to a failed container creation based on the schemaVersion 1 image. This prevented new deployment creations. With this release, the image digest calculation is fixed and the new Operators can be installed. (link:https://issues.redhat.com/browse/OCPBUGS-55435[OCPBUGS-55435])

Reviewer 2: ```suggestion
* Previously, there was an issue in the image reference digest calculation that led to a failed container creation based on the Schema version 1 image. This prevented new deployment creations. With this release, the image digest calculation is fixed and the new Operators can be installed. (link:https://issues.redhat.com/browse/OCPBUGS-55435[OCPBUGS-55435])
```

> * Previously, {azure-first} Spot Virtual Machines (VMs) that were evicted before their node became ready, could get stuck in the 'provisioned' state. With this release, {azure-short} Spot VMs now use a delete eviction policy, ensuring that the VMs correctly transition to the 'failed' state if they are preempted. (link:https://issues.redhat.com/browse/OCPBUGS-55373[OCPBUGS-55373])

Reviewer 2: ```suggestion
* Previously, {azure-first} Spot Virtual Machines (VMs) that were evicted before their node became ready could get stuck in the `provisioned` state. With this release, {azure-short} Spot VMs now use a delete eviction policy, ensuring that the VMs correctly transition to the `failed` state if they are preempted. (link:https://issues.redhat.com/browse/OCPBUGS-55373[OCPBUGS-55373])
```

> * Previously, Azure spot machines that evicted before their node became ready, could get stuck in the 'provisioned' state. With this release, Azure spot instances now use a delete eviction policy, ensuring the machines correctly transition to the 'failed' state upon preemption. (link:https://issues.redhat.com/browse/OCPBUGS-55373[OCPBUGS-55373])

Reviewer: ```suggestion
* Previously, {azure-first} noted machines that were evicted before their node was ready,  and the nodes could get stuck in the 'provisioned' state. With this release, Azure instances use a delete eviction policy, ensuring the machines correctly transition to the `failed` state upon preemption. (link:https://issues.redhat.com/browse/OCPBUGS-55373[OCPBUGS-55373])
```

Reviewer 2: ```suggestion
* Previously, {azure-first} Spot Virtual Machines (VMs) that were evicted before their node became ready could get stuck in the `provisioned` state. With this release, {azure-short} Spot VMs now use a delete eviction policy, ensuring that the VMs correctly transition to the `failed` state if they are preempted. (link:https://issues.redhat.com/browse/OCPBUGS-55373[OCPBUGS-55373])
```

> * Previously, the `oc-mirror` plugin returned an exit status of `0` (success) even when mirroring errors occurred in automated workflows. As a result, customers could not rely on the exit status in the automated workflows. With this release, the `oc mirror` returns a non-zero (failure) exit status when there are errors. (link:https://issues.redhat.com/browse/OCPBUGS-54626[OCPBUGS-54626])

Reviewer: ```suggestion
* Previously, the `oc-mirror` plugin returned an exit status of `0` (success) even when mirroring errors occurred in automated workflows. As a result, customers could not rely on the exit status in the automated workflows. With this release, the `oc mirror` returns a null (failure) exit status when there are errors. (link:https://issues.redhat.com/browse/OCPBUGS-54626[OCPBUGS-54626])
```
See ISG: https://www.ibm.com/docs/en/ibm-style?topic=word-usage#n

Author: Done

> * Previously, the `oc-mirror` plugin returned an exit status of `0` (success) even when mirroring errors occurred in automated workflows. As a result, customers could not rely on the exit status in the automated workflows. With this release, the `oc mirror` returns a null (failure) exit status when there are errors. (link:https://issues.redhat.com/browse/OCPBUGS-54626[OCPBUGS-54626])

Reviewer 2: ```suggestion
* Previously, the `oc-mirror` plugin returned an exit status of `0`, indicating success, even when mirroring errors occurred in automated workflows. As a result, customers could not rely on the exit status in the automated workflows. With this release, the `oc mirror` plugin returns a null exit status, indicating failure, when there are errors. (link:https://issues.redhat.com/browse/OCPBUGS-54626[OCPBUGS-54626])
```

> * Previously, an image pull secret generated for the internal image registry wwas not regenerated until after the embedded credentials had expired. This resulted in a small period of time during which the image pull secrets were invalid. With this release, the image pull secrets are refreshed before the embedded credentials expire. (link:https://issues.redhat.com/browse/OCPBUGS-54304[OCPBUGS-54304])

Reviewer 2: ```suggestion
* Previously, an image pull secret generated for the internal image registry was not regenerated until after the embedded credentials expired. This resulted in a small period of time during which the image pull secrets were invalid. With this release, the image pull secrets are refreshed before the embedded credentials expire. (link:https://issues.redhat.com/browse/OCPBUGS-54304[OCPBUGS-54304])
```

> * Previously, an image pull secret generated for the internal Image Registry would not be regenerated until after the embedded credentials had expired This resulted in a small period of time in which the image pull secrets were invalid. With this release, the image pull secrets are refreshed before the embedded credentials have expired. (link:https://issues.redhat.com/browse/OCPBUGS-54304[OCPBUGS-54304])

Reviewer 2: ```suggestion
* Previously, an image pull secret generated for the internal image registry was not regenerated until after the embedded credentials expired. This resulted in a small period of time during which the image pull secrets were invalid. With this release, the image pull secrets are refreshed before the embedded credentials expire. (link:https://issues.redhat.com/browse/OCPBUGS-54304[OCPBUGS-54304])
```

> * Previously, the 4.18 machine-config operators were unaware that the package-based Red Hat Enterprise Linux (RHEL) support was removed in 4.19. With this release,these operators will block updates to 4.19 on clusters with packaged-based RHEL nodes, ensuring compatibility. (link:https://issues.redhat.com/browse/OCPBUGS-53427[OCPBUGS-53427])

Reviewer: ```suggestion
* Previously, the 4.18 Machine Config Operators (MCO) were unaware that the package-based Red Hat Enterprise Linux (RHEL) support was removed in 4.19. With this release, these Operators block updates to 4.19 on clusters with packaged-based RHEL nodes, ensuring compatibility. (link:https://issues.redhat.com/browse/OCPBUGS-53427[OCPBUGS-53427])
```

> * Previously, the 4.18 Machine Config Operators (MCO) were unaware that the package-based Red Hat Enterprise Linux (RHEL) support was removed in 4.19. With this release,these Operators will block updates to 4.19 on clusters with packaged-based RHEL nodes, ensuring compatibility. (link:https://issues.redhat.com/browse/OCPBUGS-53427[OCPBUGS-53427])

Reviewer 2: ```suggestion
* Previously, the Machine Config Operator (MCO) in {product-title} 4.18 was not updated to reflect that package-based {op-system-base-full} support was removed in {product-title} 4.19. With this release, this Operator ensures compatibility by blocking updates to {product-title} 4.19 on clusters with packaged-based {op-system-base} nodes. (link:https://issues.redhat.com/browse/OCPBUGS-53427[OCPBUGS-53427])
```
